src/Typist.jsx

```diff
@@ -115,6 +129,45 @@ export default class Typist extends Component {
       });
     });
   }
+  removeAllLines() {
+    let lines = this.state.text;
+    if (lines.length > 1) {
+      lines = lines.slice(0, lines.length - 1);
+      this.setState({ text: lines,
+                            isDone: false,
+                         });
+    } else {
+      this.setState({ isDone: false });
+    }
+    return utils.eachPromiseRemove(lines, (line, idx) => {
+      if (!this.mounted) { return Promise.resolve(); }
+      return new Promise((resolve) => {
+        this.setState({ flag: !this.state.flag }, () => {
```

Comment:
hmm I'm not sure why this is necessary. Do you want to defer the execution of `this.removeLine`? Why do we need to set this state which isn't used anywhere to run `removeLine`?

I removed the `flag` state and it behaves the same way